src/Illuminate/Foundation/Auth/AuthenticatesUsers.php

```diff
@@ -106,8 +107,13 @@ protected function sendLoginResponse(Request $request)
 
         $this->clearLoginAttempts($request);
 
-        return $this->authenticated($request, $this->guard()->user())
-                ?: redirect()->intended($this->redirectPath());
+        if ($response = $this->authenticated($request, $this->guard()->user())) {
+            return $response;
+        }
+
+        return $request->wantsJson()
+                    ? new Response
```

Comment:
any reason why we are returning `empty` responses here?  :)

Reply:
What else would we return?

Reply:
how would we go about if we need to customize `responses` for a given client? - I would expect the ability to hook into this process and return whatever is needed. 

I am probably missing the whole picture here, but I got confused.

Reply:
@taylorotwell Maybe as the response has no content shouldn't  we return it with a 204 status code (No Content)?

@gocanto That is what the `authenticated` method is for. You can override it in your app and it would return a custom response.

Reply:
+1 for 204 No Content response if an empty response is being returned.

Reply:
204 would indeed seem appropriate for these responses: https://httpstatuses.com/204